components/channel_selector_modal/channel_selector_modal.tsx

```diff
@@ -131,18 +137,22 @@ export default class ChannelSelectorModal extends React.PureComponent {
         }
     }
 
-    handleDelete = (values) => {
+    handleDelete = (values: ChannelWithTeamData[]) => {
         this.setState({values});
     }
 
-    search = (term, multiselectComponent) => {
+    search = (term: string, multiselectComponent: MultiSelect<ChannelWithTeamData & Value>) => {
         if (multiselectComponent.state.page !== 0) {
             multiselectComponent.setState({page: 0});
         }
         this.props.actions.setModalSearchTerm(term);
     }
 
-    renderOption = (option, isSelected, onAdd, onMouseMove) => {
+    renderOption = (
+        option: ChannelWithTeamData & Value,
+        isSelected: boolean,
+        onAdd: (value: ChannelWithTeamData & Value) => void,
+        onMouseMove: (value: ChannelWithTeamData & Value) => void) => {
```

Comment:
`ChannelWithTeamData & Value` can be replaced with the type definition for that intersection here and all occurrences in this file. See comment below.

Reply:
I've updated the PR with this suggestion.